modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3769,6 +3769,27 @@ void GDScriptAnalyzer::reduce_identifier_from_base(GDScriptParser::IdentifierNod
 		}
 	}
 
+	// Check non-GDScript scripts.
+	const Ref<Script> script_type = base.script_type;
```

Comment:
```suggestion
	Ref<Script> script_type = base.script_type;
```
Unnecessary here and potentially misleading, all it does is prevent reassigning it with a different value, `Ref` allows you to operate on it regardless